### PR TITLE
Add configurable ingress, rename all files to `.yaml` for conformity.

### LINKED ISSUE #1: Ingress Template and Configuration
### 📜 Description

I couldn't get Ingress working. I can also not see any ingress template but it is in values.yaml. Should this already be working.

### 👟 Reproduction steps

ingress:
  enabled: true
  className: 'alb'
  annotations:
    external-dns.alpha.kubernetes.io/hostname: xxxx
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
    alb.ingress.kubernetes.io/group.name: xxxx
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP":80},{"HTTPS": 443}]'
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}'
  hosts:
    - host: xxxx
      paths:
        - path: /
          pathType: Prefix
  tls: []

### 👍 Expected behavior

ingress should be created

### 👎 Actual Behavior

no ingress created

### ☸️ Kubernetes Version

v1.29.7

### ⎈ Helm Version

v3.15.4

### 📊 Chart Version

1.0.1.

### ☁️ Cloud Provider

AWS

### 🛠️ Custom Values

_No response_

### 📃 Additional Context

_No response_

### 👀 Have you checked if this bug has been reported before?

- [X] I have searched existing issues and didn't find a similar bug report

### Are you willing to submit a PR?

Yes, I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by captainswain**
## Description
This PR updates the Ingress configuration in the Postiz Helm chart. It improves the flexibility and usability of the Ingress resource, allowing users to more easily configure their Ingress settings. The README has also been updated to reflect these changes and provide better guidance on using custom value files. I also renamed all `.yml` files to `.yaml` for conformity and best practices.

Fixes #1 

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?
- [x] Tested Ingress configuration with default values
- [x] Tested Ingress configuration with custom values

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Additional Information
- **Does this change introduce any new Kubernetes resource types?** Yes
- **Does this change modify any existing Kubernetes resource types?** Yes
  If yes, please describe the modifications: Updated the Ingress resource to be more configurable
- **Are there any changes to the values.yaml file?** Yes
  If yes, please describe the changes and their purpose: Added new Ingress configuration options to allow for more flexible setup
- **Does this change require any specific Kubernetes permissions or RBAC changes?** No
- **Are there any changes to chart dependencies?** No
- **Have you updated the CHANGELOG.md file?** No

## Additional context
This PR aims to improve the usability of the Postiz Helm chart by providing more flexible Ingress configuration options. It also enhances the documentation to guide users on how to use custom value files for their deployments.

### CODE REVIEW

**Comment by jamesread** [comment]
Hey @captainswain , this LGTM. Thanks very much for your first commit on this project.

Next time, please try to do "one issue per PR" - renaming .yml to .yaml and adding an ingress are not related changes. However, I'll allow it this time to keep things simple.

You have a minor lint failure in Chart.yaml, with trailing spaces on line 3. Please could you patch this with an additional commit in this same PR to make the CI job run successfully. Thanks https://github.com/gitroomhq/postiz-helmchart/actions/runs/11100843312/job/30964295977?pr=2

File: charts/postiz/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: postiz-app
 description: A Social Media Scheduling App 
 type: application
-version: 1.0.1
+version: 1.0.2
```
Comment (jamesread) [inline charts/postiz/Chart.yaml:5]: The trailing spaces on this line is causing the CI lint failure.

Comment (captainswain) [reply charts/postiz/Chart.yaml:5]: I removed the trailing space! Thanks

**Comment by captainswain** [comment]
@jamesread Thanks for the review I appreciate it. I'll make sure to isolate my future PRs better, it was just a chore that i ran into while getting the chart to work.


I can't personally re-run the pipeline, but we should go :+1:

**Comment by jonathanirvin-ast** [comment]
Chiming in here. Great structure and I don't have much feedback. My day job has me pegged unfortunately, but I need to bang out some GH issues and talk about the arch more.

**Comment by jonathan-irvin** [comment]
Can we isolate this into separate PRs please?

File: charts/postiz/Chart.yaml [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 appVersion: "1.3.0"
+maintainers:
+  - name: jonathan-irvin
+    email: offendingcommit@gmail.com
+    url: https://linktr.ee/offendingcommit
+  - name: jamesread
+    email: contact@jread.com
+    url: http://jread.com
```
Comment (jonathan-irvin) [inline charts/postiz/Chart.yaml:13]: I appreciate that!

File: charts/postiz/Chart.yaml [hunk 3]
```diff
@@ -1,9 +1,16 @@
 apiVersion: v2
 name: postiz-app
-description: A Social Media Scheduling App 
+description: A Social Media Scheduling App
 type: application
-version: 1.0.1
+version: 1.0.2
```
Comment (jonathan-irvin) [inline charts/postiz/Chart.yaml:5]: TODO: Make this automatic somehow

File: charts/postiz/templates/postiz-ingress.yaml
Comment (jonathan-irvin) [inline charts/postiz/templates/postiz-ingress.yaml:1]: I appreciate this! I'll need to setup some kind of environment for testing purposes.

File: charts/postiz/values.yaml [hunk 4]
```diff
@@ -25,15 +25,22 @@ service:
   port: 80
 
 ingress:
-  enabled: false
+  enabled: false  # Disabled by default
```
Comment (jonathan-irvin) [inline charts/postiz/values.yaml:28]: nit: prefer block comment here with more explanation.

File: README.md
Comment (jonathan-irvin) [inline README.md:1]: bless you for updating the docs!

**Comment by jamesread** [comment]
Hey @jonathan-irvin , I requested initially that this PR be broken up for the future (renaming .yaml and adding ingress), but then I brutalized with a bunch of commits of my own to get the CI job to work. I'm going to continue doing that until we get something working, because this is the first PR on this repo, and then subsequent commits can have stricter review.

**Review by jonathan-irvin — CHANGES_REQUESTED** [review]
All-in-all, great work. I'd appreciate it if you can do the renaming and the fixes into 2 PRs. If we have to force merge something, so be it.

**Pull request merged**